backend/src/main/java/ca/bc/gov/hlth/mohums/controller/UsersController.java

```diff
@@ -67,116 +65,72 @@ public UsersController(WebClientService webClientService, @Value("${config.vanit
             users = filteredUsers;
-            
+
             logger.debug("size for filteredUsers by org: " + filteredUsers.size());
             searchResults = ResponseEntity.status(searchResults.getStatusCode()).body(filteredUsers);
         }
-        
-        // if LastLog param are present then run the Events search to find lastLogDate for each user
-       Integer valueIteration=100;
-       if ((lastLogAfter.isPresent() || lastLogBefore.isPresent()) && !CollectionUtils.isEmpty(users)) {
-            
-            // 1st option: for every user, query all their events => NO, could 1000s of users
-            // or 2nd option: get all events for last year and filter by date and users
-        
-            logger.debug("Start of search+filter all events by user for last Log Date: " + LocalDateTime.now());
-            
-            List<Object> allEventsLastLog= new ArrayList <> ();
-            // Type=LOGIN and dateFrom=one year from now
+
+        if ((lastLogAfter.isPresent() || lastLogBefore.isPresent()) && !CollectionUtils.isEmpty(users)) {
+
+            List<LinkedHashMap<String, Object>> allEventsLastLog= new ArrayList <> ();
             LocalDate oneYearAgo = LocalDate.now().minus(1, ChronoUnit.YEARS);
             Optional<String> oneYearAgoParam = Optional.of(oneYearAgo.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")));
-            
-            MultiValueMap<String, String> queryEventLastLogParams = buildQueryEventActiveParam("0", valueIteration.toString(), oneYearAgoParam, Optional.empty());
-            List<Object> eventsLastLog = (ArrayList) webClientService.getEvents(queryEventLastLogParams).getBody();
-            
-            Integer start = 0;
-            while(eventsLastLog != null && !eventsLastLog.isEmpty()){
-            
+
+            int start = 0;
+            int maxEvents = 100;
+            List<LinkedHashMap<String, Object>> eventsLastLog;
+            do {
+                MultiValueMap<String, String> queryEventLastLogParams = buildQueryEventActiveParam(start, maxEvents, oneYearAgoParam, Optional.empty());
+                eventsLastLog = (List<LinkedHashMap<String, Object>>) (Object) webClientService.getEvents(queryEventLastLogParams).getBody();
                 allEventsLastLog.addAll(eventsLastLog);
-                
-                if(eventsLastLog.size() == valueIteration){
-                    start = start+valueIteration;
-                    queryEventLastLogParams = buildQueryEventActiveParam(start.toString(), valueIteration.toString(), oneYearAgoParam, Optional.empty());
-                    eventsLastLog = (ArrayList) webClientService.getEvents(queryEventLastLogParams).getBody();
-                } else {
-                    break;
-                }
-            }
-            logger.info("final size for AllEventsLastLog: " + allEventsLastLog.size());
+                start += maxEvents;
+            } while (!CollectionUtils.isEmpty(eventsLastLog));
```

Comment:
Using a `do while` here makes the code a bit shorter because we don't need to duplicate the `buildQueryEventActiveParam` and `getEvents` calls.

Reply:
that looks great, I'll just add back the condition on the number of events returned